[TC-5357] Support destructive alert style

hansemannn opened the pull request:
JIRA reference: https://jira.appcelerator.org/browse/TC-5357

appc-bot committed:
TC-5357: Support destructive alert style

appc-bot committed:
TC-5357: Update documentation

pec1985 commented on iphone/Classes/TiUIAlertDialogProxy.m:
```diff
@@ -134,7 +136,7 @@ -(void)show:(id)args
                 if (!IS_NULL_OR_NIL(btnName)) {
                     UIAlertAction* theAction = [UIAlertAction actionWithTitle:btnName
-                                                                        style:((curIndex == cancelIndex) ? UIAlertActionStyleCancel : UIAlertActionStyleDefault)
+                                                                        style:((curIndex == cancelIndex) ? UIAlertActionStyleCancel : (curIndex == destructiveIndex) ? UIAlertActionStyleDestructive :UIAlertActionStyleDefault)
```
This line is a bit confusing and not easy to follow due to the multiple ternary operators. If would be better to spread the out for clarity purposes, something like this:
```
UIAlertActionStyle style;
if (curIndex == cancelIndex) {
	style = UIAlertActionStyleCancel;
} else if (curIndex == destructiveIndex) {
	style = UIAlertActionStyleDestructive;
} else {
	style  = UIAlertActionStyleDefault;
}
```

pec1985 replied:
Also, if this is iOS 8 an above only, make sure you check for that with:
```
if ([TiUtils isIOS8OrGreater]) { ..... }
```

pec1985 commented on apidoc/Titanium/UI/AlertDialog.yml:
```diff
@@ -168,6 +168,14 @@ properties:
+    description: |
+        Note that this property available on iOS SDK 8 or above. 
+    type: Number
+    default: -1
```
Would be nice to specify that setting it to `-1` disables this option, or does nothing. Having a default value of `-1` without explaining what it does makes it unclear

pec1985 commented:
PR reviewed, left a couple of comments.

hansemannn commented:
Thank you! Comments are reasonable, will have a look at it!

Commit:
TC-5357: Improve code- and docs-readability

hansemannn commented:
Updated docs and code readability as well as a bounding-check like it is done with the cancel-button. I think the iOS 8 check is not necessary in this case, because the code is already in the conditional iOS 8 block to support the UIAlertController.

pec1985 commented:
One more thing... the destructive index is missing from the click event. For consistency, add it here as well `fireClickEventWithAction`:
```
NSMutableDictionary *event = [NSMutableDictionary dictionaryWithObjectsAndKeys:
                               NUMUINTEGER(indexOfAction),@"index",
                               [NSNumber numberWithInt:cancelIndex],@"cancel",
                               nil];
```

Commit:
TC-5357: Add destructive index to click event

hansemannn commented:
Added the destructive index and implemented a quick sample example code to test:

``` javascript
var win = Ti.UI.createWindow({
	backgroundColor: "#fff"
});

var button = Ti.UI.createButton({
	title: "Show destructive dialog"
});

var dialog = Ti.UI.createAlertDialog({
	title: "Proceed",
	message: "Do you really want to proceed?",
	destructive: 0,
	cancel: 1,
	buttonNames: ["Destructive", "Cancel", "Proceed"]
});

button.addEventListener("click", function() {
	dialog.show();
});

dialog.addEventListener("click", function(e) {
	Ti.API.info(e);
});

win.add(button);

win.open();
```

pec1985 commented:
i have opened a new PR with your commits for Titanium master branch #6842

pec1985 commented:
There is one minor thing missing in the documentation. The `since` tag (for Ti SDK) and the `osver` for the iOS version. You can simple cherry pick from master since I already merged that
```
git cherry-pick cfcbc96
```

hansemannn commented:
Closing this outdated PR in favor of #6842
